src/NUnitEngine/nunit.engine.core/Drivers/NUnitFrameworkDriver.cs

```diff
@@ -0,0 +1,157 @@
+
+            if (nunitRef.Version >= MINIMUM_NUNIT_VERSION)
+            {
+                API = "2018";
+                _api = (NUnitFrameworkApi)testDomain.CreateInstanceFromAndUnwrap(
+                    Assembly.GetExecutingAssembly().Location,
+                    "NUnit.Engine.Drivers.NUnitFrameworkApi2018",
+                    false,
+                    0,
+                    null,
+                    new object[] { ID, nunitRef },
+                    null,
+                    null).ShouldNotBeNull();
+            }
+            else
+            {
+                API = "2009";
+                _api = new NUnitFrameworkApi2009(testDomain, ID, nunitRef);
+            }
+        }
+
+        /// <summary>
+        /// Internal generic constructor used by our tests.
+        /// </summary>
+        /// <param name="testDomain">The application domain in which to create the FrameworkController</param>
+        /// <param name="nunitRef">An AssemblyName referring to the test framework.</param>
+        internal NUnitFrameworkDriver(AppDomain testDomain, string api, string id, AssemblyName nunitRef)
+        {
+            Guard.ArgumentNotNull(testDomain, nameof(testDomain));
+            Guard.ArgumentNotNull(api, nameof(api));
+            Guard.ArgumentValid(api == "2009" || api == "2018", $"Invalid API specified: {api}", nameof(api));
+            Guard.ArgumentNotNullOrEmpty(id, nameof(id));
+            Guard.ArgumentNotNull(nunitRef, nameof(nunitRef));
+
+            ID = id;
+            API = api;
+
+            _api = api == "2018"
+                ? (NUnitFrameworkApi)testDomain.CreateInstanceFromAndUnwrap(
+                    Assembly.GetExecutingAssembly().Location,
+                    typeof(NUnitFrameworkApi2018).FullName!,
+                    false,
+                    0,
+                    null,
+                    new object[] { ID, nunitRef },
+                    null,
+                    null).ShouldNotBeNull()
+                : new NUnitFrameworkApi2009(testDomain, ID, nunitRef);
+        }
+#else
+        /// <summary>
+        /// Construct an NUnitFrameworkDriver
+        /// </summary>
+        /// <param name="reference">An AssemblyName referring to the test framework.</param>
+        public NUnitFrameworkDriver(string id, AssemblyName nunitRef)
+        {
+            Guard.ArgumentNotNullOrEmpty(id, nameof(id));
+            Guard.ArgumentNotNull(nunitRef, nameof(nunitRef));
+
+            ID = id;
+            API = "2018";
+
+            _api = new NUnitFrameworkApi2018(ID, nunitRef);
+        }
+#endif
+
+        /// <summary>
+        /// String naming the API in use, for use by tests
+        /// </summary>
+        internal string API { get; } = string.Empty;
+
+        /// <summary>
+        /// An id prefix that will be passed to the test framework and used as part of the
+        /// test ids created.
+        /// </summary>
+        public string ID { get; } = string.Empty;
```

Comment:
Suggest to drop the ` = string.Empty;` so any future change will still pass in `id` in the constructor.